### PR TITLE
FLOW-16441 New methods in nylas service for shared email accounts

### PR DESCRIPTION
**Pull request by dpotera**
## Summary

New in nylas service for shared email accounts

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] My changes generate no new warnings (ESLint)

## Links

-   Jira:

Thanks so much for your PR, your contribution is appreciated! ❤️

### CODE REVIEW

**Review by JakubJagosz — APPROVED** [review]

File: src/service/NylasService/NylasService.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import { APIClient, APIMapping, ApiResponse } from '../../http';
 import { NylasServiceTypes } from './NylasService.Types';
 import SchedulerPage = NylasServiceTypes.SchedulerPage;
 import RunningMailAccountResponse = NylasServiceTypes.RunningMailAccountResponse;
+import {SharedAccountsRequest} from "@flowfact/types/src/Nylas";
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:14]:
```suggestion
import {SharedAccountsRequest} from "@flowfact/types";
```

File: src/service/NylasService/NylasService.ts [hunk 2]
```diff
@@ -272,6 +273,30 @@ export class NylasService extends APIClient {
     async addManualAccount(account: AccountInfo) {
         return await this.invokeApiWithErrorHandling('/manual-account', 'POST', account);
     }
+
+    /**
+     * Creates shared accounts entry for given email account
+     * @param sharedAccounts
+     */
+    async createSharedAccountsList(sharedAccounts: SharedAccountsRequest) {
+        return await this.invokeApiWithErrorHandling('/shared-accounts', 'POST', sharedAccounts);
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:282]: No response type?

File: src/service/NylasService/NylasService.ts [hunk 3]
```diff
@@ -272,6 +273,30 @@ export class NylasService extends APIClient {
...
+    }
+
+    /**
+     * Updates shared accounts list by overwriting all entries
+     * @param sharedAccounts
+     */
+    async updateSharedAccountsList(sharedAccounts: SharedAccountsRequest) {
+        return await this.invokeApiWithErrorHandling('/shared-accounts', 'PUT', sharedAccounts);
```
Comment (ffjanhoeck) [inline src/service/NylasService/NylasService.ts:290]: No response type?

**Review by ffjanhoeck — CHANGES_REQUESTED** [review]

File: src/service/NylasService/NylasService.ts (hunk 1 above)
Comment (dpotera) [reply src/service/NylasService/NylasService.ts]: done

File: src/service/NylasService/NylasService.ts (hunk 2 above)
Comment (dpotera) [reply src/service/NylasService/NylasService.ts]: yes, we use just http status, added <void> type

File: src/service/NylasService/NylasService.ts (hunk 3 above)
Comment (dpotera) [reply src/service/NylasService/NylasService.ts]: as above

**Review by ffjanhoeck — APPROVED** [review]

**Pull request merged**